backend/errors/httperror_test.go

```diff
@@ -0,0 +1,64 @@
+package errors
+
+import (
+	"bytes"
+	"errors"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestHandleHTTPError(t *testing.T) {
+	tests := []struct {
+		testName       string
+		w              *httptest.ResponseRecorder
+		err            error
+		statusCode     int
+		userMessage    string
+		output         string
+		wrappedMessage string
+		res            bool
+	}{
+		{
+			testName:       "Nil",
+			w:              httptest.NewRecorder(),
+			err:            nil,
+			statusCode:     http.StatusOK,
+			userMessage:    "",
+			output:         "",
+			wrappedMessage: "",
+			res:            false,
+		},
+		{
+			testName:       "Server Error 500",
+			w:              httptest.NewRecorder(),
+			err:            errors.New("foo"),
+			statusCode:     http.StatusInternalServerError,
+			userMessage:    "bar",
+			output:         "Error: foo",
+			wrappedMessage: "{\"error\":\"bar\"}",
+			res:            true,
+		},
+	}
+	for _, test := range tests {
```

Comment:
if you put all of the code in this loop in a `t.Run()` block [go doc](https://pkg.go.dev/testing#T.Run) you can have the test results organized by test name, which will make the output more readable for table driven tests.

Reply:
reference example for usage: https://go.dev/wiki/TableDrivenTests